test/jdk/javax/swing/ToolTipManager/bug5078214.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright (c) 2004, 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 5078214
+ * @summary ToolTip is shown partially when the application is near the bottom of screen.
+ * @library ../regtesthelpers
+ * @build Util JRobot
+ * @key headful
+ * @run main bug5078214
+ */
+
+import java.awt.BorderLayout;
+import java.awt.event.MouseEvent;
+import java.awt.GraphicsConfiguration;
+import java.awt.GraphicsDevice;
+import java.awt.GraphicsEnvironment;
+import java.awt.Insets;
+import java.awt.Point;
+import java.awt.Rectangle;
+import java.awt.Toolkit;
+import java.awt.Window;
+import javax.swing.JButton;
+import javax.swing.JFrame;
+import javax.swing.SwingUtilities;
+import javax.swing.ToolTipManager;
+
+public class bug5078214 {
+    static volatile boolean passed = false;
+
+    static volatile JFrame mainFrame;
+    static volatile Rectangle bounds;
+    static volatile Insets insets;
+    static volatile JRobot r;
+
+    public static void main(String[] args) throws Exception {
+        try {
+            SwingUtilities.invokeAndWait(() -> {
+                GraphicsEnvironment ge = GraphicsEnvironment.
+                        getLocalGraphicsEnvironment();
+                GraphicsDevice[] gs = ge.getScreenDevices();
+                GraphicsConfiguration testGraphics = null;
```

Comment:
```suggestion
                GraphicsConfiguration testGraphicsConfig = null;
```
The name `testGraphics` suggests you're about to test `Graphics` object.

I would suggest moving the code which determines the `GraphicsConfiguration` to test into a helper method. And this helper method can safely be run on the main thread.